pkg/k8shandler/clusterrestart.go

```diff
@@ -280,26 +293,26 @@ func (clusterRestart ClusterRestart) ensureClusterHealthValid() error {
 func (clusterRestart ClusterRestart) requiredSetPrimariesShardsAndFlush() error {
 	// set shard allocation as primaries
 	if ok, err := clusterRestart.client.SetShardAllocation(api.ShardAllocationPrimaries); !ok {
-		return fmt.Errorf("Unable to set shard allocation to primaries: %v", err)
+		return ErrUnableToSetShardAllocations.Wrap(err, "unable to set shard allocation to primaries",
+			"namespace", clusterRestart.clusterNamespace,
+			"cluster", clusterRestart.clusterName)
 	}
 
 	// flush nodes
 	if ok, err := clusterRestart.client.DoSynchronizedFlush(); !ok {
-		log.Error(err, "Unable to perform synchronized flush", "namespace", clusterRestart.clusterNamespace, "name", clusterRestart.clusterName)
+		return ErrFlushShardsFailed.Wrap(err,
```

Comment:
this isn't a required operation... it is okay if it fails, which is why we didn't return before